lib/NetworkManager.js

```diff
@@ -178,15 +174,13 @@ class NetworkManager extends EventEmitter {
       }).catch(debugError);
     }
 
-    const requestHash = generateRequestHash(event.request);
-    const requestId = this._requestHashToRequestIds.firstValue(requestHash);
-    if (requestId) {
+    const requestId = event.requestId;
+    if (requestId && this._requestIdToRequestWillBeSentEvent.has(requestId)) {
       const requestWillBeSentEvent = this._requestIdToRequestWillBeSentEvent.get(requestId);
       this._onRequest(requestWillBeSentEvent, event.interceptionId);
-      this._requestHashToRequestIds.delete(requestHash, requestId);
```

Comment:
```js
this._requestIdToInterceptionId.delete(requestId);
```

Reply:
The requestId gets deleted out of this._requestIdToInterceptionId in _onRequestWillBeSent. If I put another .delete here it would never do anything because this branch is for when we already had a requestWillBeSent and _requestIdToInterceptionId is not used.